tools/utillities.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import os
+import sys
+import platform
+import argparse
+import shutil
+import subprocess
+
+PyautoguiAvailable = True
+try:
+    import pyautogui
+except:
+    PyautoguiAvailable = False
+
+
+class FatalError( Exception ):
+    def __init__( self, msg ):
+        super().__init__( msg )
+
+class ProcessError( Exception ):
+    def __init__( self, msg ):
+        super().__init__( msg )
+
+def Msg( txt ):
+    sys.stderr.write( txt + "\n" )
+
+def Debug( txt ):
+    sys.stderr.write( txt + "\n" )
+
+def Which_Chdir_( prg, dir ):
+    cwd = os.getcwd()
+    try:
+        os.chdir( dir )
+    except:
+        return None
+    res = None
+    try:
+        res = shutil.which( "./%s" % prg )
+        if res is not None:
+            res = os.path.abspath( os.path.join( os.getcwd(), prg ) )
+    finally:
+        os.chdir( cwd )
+    return res
+
+def Which( prg, dir = None ):
+    """ returns full path to executable 'prg'.
+        Takes care of OS, on Windows searches for <prg>.exe instead of <prg>.
+        if <dir> is given prefers prg inside this dir over others"""
+    if platform.system() == "Windows":
+        prg = "%s.exe" % prg
+    res = None
+    if dir is not None:
+        res = Which_Chdir_( prg, dir )
+    if res is None:
+        res = shutil.which( prg )
+    return res
+
+def CopyToDir( name, ext, dstDir ):
+    """ if file './name.ext' exist, copy it to 'dstDir/name.ext'. Return name of copied file as list. """
+    fileName = "%s.%s" % ( name, ext )
+    srcPath = os.path.abspath( fileName )
+    if not os.path.exists( srcPath ):
+        return []
+    shutil.copyfile( srcPath, os.path.join( dstDir, fileName ) )
+    return [ fileName ]
+
+def IntValFromStr( strVal ):
+    """ Convert string into integer, can be used with decimal value '123' or hex '0xabc'. """
+    return int( strVal.strip(), 0 )
+
+def RemoveIno( path ):
+    """ Returns the given path with extension ".ino" removed (if exist) """
+    if path[ -4: ] == ".ino":
+        path = name[ 0:-4 ]
+    return path
+
+def TkDialog( title, text, mbName ):
+    try:
```

Comment:
Is it a leftover ?

Reply:
No, it is the base of (for ex.) ConfirmDialog & WarningDialog which are used inside postbuild.py

Reply:
Anything with python must be cross-compatible with all archs.
A portable version of python is shipped by the installer and I doubt tkinter will work in that case.